android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUITableView.java

```diff
@@ -1,145 +1,61 @@
 import ti.modules.titanium.ui.widget.searchbar.TiUISearchBar;
 import ti.modules.titanium.ui.widget.searchview.TiUISearchView;
-import ti.modules.titanium.ui.widget.tableview.TableViewModel;
 import ti.modules.titanium.ui.widget.tableview.TiTableView;
-import ti.modules.titanium.ui.widget.tableview.TiTableView.OnItemClickedListener;
-import ti.modules.titanium.ui.widget.tableview.TiTableView.OnItemLongClickedListener;
-import android.app.Activity;
-import android.os.Build;
-import android.os.Bundle;
-import android.view.Gravity;
-import android.view.View;
-import android.view.ViewParent;
-import android.widget.ListView;
-import android.widget.RelativeLayout;
 
-public class TiUITableView
-	extends TiUIView implements OnItemClickedListener, OnItemLongClickedListener, OnLifecycleEvent
+public class TiUITableView extends TiUIView
 {
 	private static final String TAG = "TitaniumTableView";
 
 	private static final int SEARCHVIEW_ID = View.generateViewId();
 
-	protected TiTableView tableView;
+	protected final TiTableView tableView;
+	private TiViewProxy searchProxy;
 
 	public TiUITableView(TiViewProxy proxy)
 	{
 		super(proxy);
+
 		getLayoutParams().autoFillsHeight = true;
 		getLayoutParams().autoFillsWidth = true;
-	}
-
-	@Override
-	public void onClick(KrollDict data)
-	{
-		proxy.fireEvent(TiC.EVENT_CLICK, data);
-	}
-
-	@Override
-	public boolean onLongClick(KrollDict data)
-	{
-		return proxy.fireEvent(TiC.EVENT_LONGCLICK, data);
-	}
-
-	public void setModelDirty()
-	{
-		tableView.getTableViewModel().setDirty();
-		getTableView().dataSetChanged();
-	}
-
-	public TableViewModel getModel()
-	{
-		return tableView.getTableViewModel();
-	}
-
-	public void updateView()
-	{
-		tableView.dataSetChanged();
-	}
-
-	public void scrollToIndex(final int index)
-	{
-		tableView.getListView().setSelection(index);
-	}
 
-	public void scrollToTop(final int index)
-	{
-		tableView.getListView().setSelectionFromTop(index, 0);
-	}
-
-	public void selectRow(final int row_id)
-	{
-		tableView.getListView().setSelection(row_id);
+		this.tableView = new TiTableView((TableViewProxy) proxy);
+		setNativeView(tableView);
 	}
 
 	public TiTableView getTableView()
 	{
-		return tableView;
-	}
-
-	public ListView getListView()
-	{
-		return tableView.getListView();
+		return this.tableView;
 	}
 
 	@Override
 	public void processProperties(KrollDict d)
 	{
-		// Don't create a new table view if one already exists
-		if (tableView == null) {
-			tableView = new TiTableView((TableViewProxy) proxy);
-		}
-		Activity activity = proxy.getActivity();
-		if (activity instanceof TiBaseActivity) {
-			((TiBaseActivity) activity).addOnLifecycleEventListener(this);
-		}
-
-		boolean clickable = true;
-		if (d.containsKey(TiC.PROPERTY_TOUCH_ENABLED)) {
-			clickable = TiConvert.toBoolean(proxy.getProperty(TiC.PROPERTY_TOUCH_ENABLED), true);
-		}
-		if (clickable) {
-			tableView.setOnItemClickListener(this);
-			tableView.setOnItemLongClickListener(this);
-		}
 
```

Comment:
TODO: Both `PROPERTY_HEADER_DIVIDERS_ENABLED` and `PROPERTY_FOOTER_DIVIDERS_ENABLED` are not supported anymore. I just realized I will need to implement this into the template.